webapp/channels/src/components/convert_gm_to_channel_modal/convert_gm_to_channel_modal.tsx

```diff
@@ -0,0 +1,220 @@
+            if (response.error || !response.data) {
+                return;
+            }
+            const teams = response.data;
+
+            const teamsById: {[id: string]: Team} = {};
+            teams.forEach((team) => {
+                teamsById[team.id] = team;
+            });
+            setCommonTeamsById(teamsById);
+            setCommonTeamsFetched(true);
+
+            // if there is only common team, selected it.
+            if (teams.length === 1) {
+                setSelectedTeamId(teams[0].id);
+            }
+        };
+
+        work();
+        setTimeout(() => setLoadingAnimationTimeout(true), 1200);
+    }, []);
+
+    const handleConfirm = async () => {
+        if (!selectedTeamId) {
+            return;
+        }
+
+        const {error} = await props.actions.convertGroupMessageToPrivateChannel(props.channel.id, selectedTeamId, channelName.trim(), channelURL.trim());
+
+        if (error) {
+            setConversionError(error.message);
+            return;
+        }
+
+        setConversionError(undefined);
+        if (props.channelsCategoryId) {
+            props.actions.moveChannelsInSidebar(props.channelsCategoryId, 0, props.channel.id, false);
+        }
+        trackEvent('actions', 'convert_group_message_to_private_channel', {channel_id: props.channel.id});
+        props.onExited();
+    };
+
+    const showLoader = !commonTeamsFetched || !loadingAnimationTimeout;
+
+    const canCreate = (): boolean => {
+        return selectedTeamId !== undefined && channelName !== '' && !nameError;
+    };
+
+    const modalProps: Partial<ComponentProps<typeof GenericModal>> = {};
+    let modalBody;
+
+    if (!showLoader && Object.keys(commonTeamsById).length === 0) {
+        modalProps.confirmButtonText = formatMessage({id: 'generic.okay', defaultMessage: 'Okay'});
+        modalProps.handleConfirm = props.onExited;
+
+        modalBody = (
+            <div className='convert-gm-to-channel-modal-body error'>
+                <NoCommonTeamsError/>
+            </div>
+        );
+    } else {
+        modalProps.handleCancel = showLoader ? undefined : props.onExited;
+        modalProps.isDeleteModal = true;
+        modalProps.cancelButtonText = formatMessage({id: 'channel_modal.cancel', defaultMessage: 'Cancel'});
+        modalProps.confirmButtonText = formatMessage({id: 'sidebar_left.sidebar_channel_modal.confirmation_text', defaultMessage: 'Convert to private channel'});
+        modalProps.isConfirmDisabled = !canCreate();
+
+        let subBody;
+        if (showLoader) {
+            subBody = (
+                <div className='loadingIndicator'>
+                    <LoadingSpinner/>
+                </div>
+            );
+        } else {
+            subBody = (
+                <React.Fragment>
+                    <WarningTextSection channelMemberNames={channelMemberNames}/>
+
+                    {
+                        Object.keys(commonTeamsById).length > 1 &&
```

Comment:
I wonder if we should filter the `commonTeamsById` by those where you can create private channels on. If so, we should update the error message to also consider that possibility.

Reply:
Noted for follow up PR. Makes sense.